packages/h5p-server/src/ContentUserDataManager.ts

```diff
@@ -0,0 +1,154 @@
+import {
+    ContentId,
+    IContentUserData,
+    IUser,
+    IContentUserDataStorage
+} from './types';
+
+import Logger from './helpers/Logger';
+
+const log = new Logger('ContentUserDataManager');
+
+/**
+ * The ContentUserDataManager takes care of saving user data and states. It only contains storage-agnostic functionality and
+ * depends on a ContentUserDataStorage object to do the actual persistence.
+ */
+export default class ContentUserDataManager {
+    /**
+     * @param contentUserDataStorage The storage object
+     */
+    constructor(contentUserDataStorage: IContentUserDataStorage) {
+        log.info('initialize');
+        this.contentUserDataStorage = contentUserDataStorage;
+    }
+
+    private contentUserDataStorage: IContentUserDataStorage;
+
+    /**
+     * Deletes a contentUserData object.
+     * Throws errors if something goes wrong.
+     * @param contentId The content id to delete.
+     * @param user The user who wants to delete the content (not the user the contentUserData belongs to)
+     */
+    public async deleteContentUserData(
+        contentId: ContentId,
+        user: IUser
+    ): Promise<void> {
+        if (this.contentUserDataStorage) {
+            log.info(`deleting contentUserData for ContentId ${contentId}`);
+            return this.contentUserDataStorage.deleteContentUserData(
+                contentId,
+                user
+            );
+        }
+        return;
+    }
+
+    /**
+     * Loads the contentUserData for given contentId, dataType and subContentId
+     * @param contentId The id of the content to load user data from
+     * @param dataType Used by the h5p.js client
+     * @param subContentId The id provided by the h5p.js client call
+     * @param user The user who owns this object
+     * @returns the saved state as string
+     */
+    public async loadContentUserData(
+        contentId: ContentId,
+        dataType: string,
+        subContentId: string,
+        user: IUser
+    ): Promise<string> {
+        if (!this.contentUserDataStorage) {
+            return undefined;
+        }
+
+        if (!this.contentUserDataStorage.loadContentUserData) {
```

Comment:
Is this really necessary? An implementation wouldn't make much sense without this method, as it's not optional. We don't do this is the library anywhere else.

Reply:
ok, I will take it out.